contracts/helpers/Harvester.sol

```diff
@@ -0,0 +1,185 @@
+    uint64 targetExposure;
+    // Maximum exposure within the Transmuter to the vault asset
+    uint64 maxExposureYieldAsset;
+    // Minimum exposure within the Transmuter to the vault asset
+    uint64 minExposureYieldAsset;
+    // Whether limit exposures should be overriden or read onchain through the Transmuter
+    uint64 overrideExposures;
+}
+
+/// @title Harvester
+/// @author Angle Labs, Inc.
+/// @dev Contract for anyone to permissionlessly adjust the reserves of Angle Transmuter through
+/// the RebalancerFlashloan contract
+contract Harvester is AccessControl {
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+
+    /// @notice Reference to the `transmuter` implementation this contract aims at rebalancing
+    ITransmuter public immutable TRANSMUTER;
+    /// @notice Permissioned rebalancer contract
+    RebalancerFlashloan public rebalancer;
+    /// @notice Max slippage when dealing with the Transmuter
+    uint96 public maxSlippage;
+    /// @notice Data associated to a collateral
+    mapping(address => CollatParams) public collateralData;
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                    INITIALIZATION                                                  
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    constructor(
+        address _rebalancer,
+        address vault,
+        uint64 targetExposure,
+        uint64 overrideExposures,
+        uint64 maxExposureYieldAsset,
+        uint64 minExposureYieldAsset,
+        uint96 _maxSlippage
+    ) {
+        ITransmuter transmuter = RebalancerFlashloan(_rebalancer).TRANSMUTER();
+        TRANSMUTER = transmuter;
+        rebalancer = RebalancerFlashloan(_rebalancer);
+        accessControlManager = IAccessControlManager(transmuter.accessControlManager());
+        _setCollateralData(vault, targetExposure, minExposureYieldAsset, maxExposureYieldAsset, overrideExposures);
+        _setMaxSlippage(_maxSlippage);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        HARVEST                                                     
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Invests or divests from the yield asset associated to `collateral`
+    /// @dev This transaction either reduces the exposure to `collateral` in the Transmuter or frees up some collateral
+    /// that can then be used for people looking to burn
+    /// @dev Due to potential transaction fees within the Transmuter, this function doesn't exactly bring `collateral`
+    /// to the target exposure
+    function harvest(address collateral) external {
```

Comment:
If we use this function we can't have an order for this pair on the Rebalancer contract, otherwise attacker can manipulate the exposures and empty the order for free. We should let them pay for the swaps on the transmuter